sockeye/layers.py

```diff
@@ -108,6 +112,64 @@ def __call__(self) -> mx.sym.Symbol:
                                     rhs=self.scale, name="%swn_scale" % self.prefix)
 
 
+class OutputLayer:
+    """
+    Defines the output layer of Sockeye decoders. Supports weight tying and weight normalization.
+
+    :param num_hidden: Number of hidden units in layer input (decoder representation).
+    :param num_embed: Target embedding size.
+    :param vocab_size: Target vocabulary size.
+    :param weight_tying: Whether to use target embedding parameters as output layer parameters.
+    :param embed_weight: Optional embedding matrix. Required if weight_tying == True.
+    :param weight_normalization: Whether to apply weight normalization.
+    :param prefix: Prefix used for naming.
+    """
+
+    def __init__(self,
+                 num_hidden: int,
+                 num_embed: int,
+                 vocab_size: int,
+                 weight_tying: bool,
+                 embed_weight: Optional[mx.sym.Symbol],
+                 weight_normalization: bool,
+                 prefix: str = '') -> None:
+        self.vocab_size = vocab_size
+
+        self.w = mx.sym.Variable("%scls_weight" % prefix, shape=(vocab_size, num_hidden))
```

Comment:
how about moving these into the else branch of `weight_tying` to only create it if weight tying is turned off?

Reply:
good call, will change!